src/services/ApiService.ts

```diff
@@ -239,36 +239,19 @@ const getContractCallQuote = async (
 }
 
 const getStatus = async (
-  { bridge, fromChain, toChain, txHash }: GetStatusRequest,
+  requestConfig: GetStatusRequest,
   options?: RequestOptions
 ): Promise<StatusResponse> => {
-  if (fromChain !== toChain && !bridge) {
-    throw new ValidationError(
-      'Parameter "bridge" is required for cross chain transfers.'
-    )
-  }
-
-  if (!fromChain) {
-    throw new ValidationError('Required parameter "fromChain" is missing.')
-  }
-
-  if (!toChain) {
-    throw new ValidationError('Required parameter "toChain" is missing.')
-  }
-
-  if (!txHash) {
+  if (!requestConfig.txHash) {
     throw new ValidationError('Required parameter "txHash" is missing.')
   }
 
   const config = ConfigService.getInstance().getConfig()
   try {
     const response = await request<StatusResponse>(
-      `${config.apiUrl}/status?${new URLSearchParams({
-        bridge,
-        fromChain,
-        toChain,
-        txHash,
-      } as Record<string, string>)}`,
+      `${config.apiUrl}/status?${new URLSearchParams(
```

Comment:
let's move the formation of search params to a variable and then use that variable here.

Increase readability imo.

Reply:
but it is not clean, I think
```ts
const getStatus = async (
  { txHash, fromChain, toChain, bridge }: GetStatusRequest,
  options?: RequestOptions
): Promise<StatusResponse> => {
  if (!txHash) {
    throw new ValidationError('Required parameter "txHash" is missing.')
  }
  const requestConfig: GetStatusRequest = { txHash }
  if (fromChain) {
    requestConfig.fromChain = fromChain
  }
  if (toChain) {
    requestConfig.toChain = toChain
  }
  if (bridge) {
    requestConfig.bridge = bridge
  }
  const config = ConfigService.getInstance().getConfig()
  try {
    const response = await request<StatusResponse>(
      `${config.apiUrl}/status?${new URLSearchParams(
        requestConfig as unknown as Record<string, string>
      )}`,
      {
        signal: options?.signal,
      }
    )

    return response
  } catch (e) {
    throw await parseBackendError(e)
  }
}
```

Reply:
Got it.
Agreed on the removal of multiple checks for the keys.

I was referring more towards the `URLSearchParams`.

```
try {

    // added below
    const queryParams = new URLSearchParams(requestConfig as unknown as Record<string, string>);

    const response = await request<StatusResponse>(
      `${config.apiUrl}/status?${queryParams}`,
      {
        signal: options?.signal,
      }
    )

    return response
  } catch (e) {
    throw await parseBackendError(e)
  }
```